### PR TITLE
Build avx

### PR DESCRIPTION
**Pull request by gregsinclair42**
This fixes the way we linked against avx, mkl, and thread libraries. Also uses a more recent build of faiss. End result is query time is 3x faster than the 0.3.0-beta03 version and 1.4X faster than the 0.11 version.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -33,18 +33,17 @@ RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/nul
...
 
 # build faiss and the c api
 RUN apt-get -y install git && \
     git clone -b ${FAISS_VERSION} https://github.com/facebookresearch/faiss.git /faiss && \
     cd /faiss && \
-    cmake -DFAISS_ENABLE_GPU=OFF -DFAISS_ENABLE_PYTHON=OFF -DBUILD_TESTING=ON -DCMAKE_BUILD_TYPE=Release -DFAISS_ENABLE_C_API=ON -DBUILD_SHARED_LIBS=ON -DFAISS_OPT_LEVEL=avx2 -B build . && \
-    make -C build -j $(nproc) faiss && \
-    make -j $(nproc) -C build install && \
-    cd /
+    git checkout add3705c1147a51b1b20161cc6c08945bd485f14 && \
```
Comment (andyalm) [inline Dockerfile]: What is the significance of this SHA? Is there a tag for it that we could use instead?

File: build-faiss-linux.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash -e
 
-FAISS_VERSION=${1:-v1.7.1}
+FAISS_VERSION=${1:-main}
```
Comment (andyalm) [inline build-faiss-linux.sh]: Is there no release tag for the version you are wanting to use right now?

Comment (gregsinclair42) [reply build-faiss-linux.sh]: unfortunately there's no tag for this version. i can try against their latest release tag and see if it has any impact on the timings.

File: Dockerfile (hunk 1 above)
Comment (gregsinclair42) [reply Dockerfile]: i'll try using their latest release tag and see if the speedup improvements still work. agree it would be better to use a release tag.

**Pull request merged**